File: tests/xcalablemp/global-view/gmove/F/gmove_lc_2a12t_b_gb.f90
program tgmove
integer i
integer,parameter :: n=4
integer a(n,n),b(n,n)
integer m(2)=(/2,2/)
integer xmp_node_num
!$xmp nodes p(4)
!$xmp nodes q(2,2)
!$xmp template tx(n)
!$xmp template ty(n,n)
!$xmp distribute tx(block) onto p
!$xmp distribute ty(gblock(m),gblock(m)) onto q
!$xmp align a(i,*) with tx(i)
!$xmp align b(i,j) with ty(i,j)

!$xmp loop (i) on tx(i)
do j=1,n
  do i=1,n
    a(i,j)=i+j
  end do
end do

!$xmp loop (i,j) on ty(i,j)
do j=1,n
  do i=1,n
    b(i,j)=0
  end do
end do

!$xmp gmove
b(2:n,2:n)=a(2:n,2:n)

ierr=0
!$xmp loop (i,j) on ty(i,j)
do j=2,n
  do i=2,n
    ierr=ierr+abs(b(i,j)-a(i,j))
  end do
end do

!$xmp barrier
!$xmp reduction (max:ierr)
call chk_int(ierr)

stop
end program tgmove

File: tests/xcalablemp/global-view/gmove/C/gmove_lc_2a12t_b_gb.c
#include <stdio.h>
#include <stdlib.h>

extern int chk_int(int ierr);

int n=16;
int a[n][n],b[n][n];
int m1[2]={9,7};
#pragma xmp nodes p(4)
#pragma xmp nodes q(2,2)
#pragma xmp template tx(0:n-1)
#pragma xmp template ty(0:n-1,0:n-1)
#pragma xmp distribute tx(block) onto p
#pragma xmp distribute ty(gblock(m1),gblock(m1)) onto q
#pragma xmp align a[i0][*] with tx(i0)
#pragma xmp align b[i0][i1] with ty(i0,i1)

int main(){

  int i0,i1,ierr;

#pragma xmp loop (i0) on tx(i0)
  for(i0=0;i0<n;i0++){
    for(i1=0;i1<n;i1++){
      a[i0][i1]=i0+i1+1;
    }
  }

#pragma xmp loop (i0,i1) on ty(i0,i1)
  for(i0=0;i0<n;i0++){
    for(i1=0;i1<n;i1++){
      b[i0][i1]=0;
    }
  }

#pragma xmp gmove
  b[0:n][0:n]=a[0:n][0:n];

  ierr=0;
#pragma xmp loop (i0,i1) on ty(i0,i1)
  for(i0=0;i0<n;i0++){
    for(i1=0;i1<n;i1++){
      ierr=ierr+abs(b[i0][i1]-i0-i1-1);
    }
  }

#pragma xmp reduction (MAX:ierr)
  chk_int(ierr);
}
